archunit/src/main/java/com/tngtech/archunit/core/importer/TargetAtLine.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2014-2020 TNG Technology Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tngtech.archunit.core.importer;
+
+import com.tngtech.archunit.core.domain.JavaType;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+class TargetAtLine {
```

Comment:
This class looks very similar to `RawInstanceofCheck` except that the type could not be resolved yet. If you look at `RawAccessRecord` and similar classes, then your `TargetAtLine` would actually be `RawInstanceofCheck` in this terminology and your `RawInstanceofCheck` would actually be `ProcessedInstanceofCheck` (because it has been enriched with the correct info from the `ImportedClasses`).
What about actually calling this class `RawInstanceofCheck` and give it a `process(ImportedClasses classes)` method to resolve the `JavaClass` for `target`? The result could just be an nested class `Processed` (which would then naturally reside in the `importer` package and thus solve my comment above). When traversing the border from the `importer` package to the `domain` package we could then just pass `JavaClass target` and `int lineNumber` as separate parameters and the `domain` package would not be influenced by `RawInstanceofCheck` and `RawInstancecofCheck.Processed`. What do you think?

I.e.

```
class RawInstanceofCheck {
    private final JavaType target;
    private final int lineNumber;

    // ...

    Processed process(ImportedClasses classes) {
        // convert JavaType -> JavaClass via classes.get(..)
    }

    // ...

    static class Processed {
        private final JavaClass target;
        private final int lineNumber;
    }
}
```

Reply:
Good point. Thanks. 😃 
I removed to old `RawInstanceofCheck` and renamed `TargetAtLine` to `RawInstanceofCheck`. The inner class `Processed` does not seem to be necessary anymore.